dashboard/server/api/index.js

```diff
@@ -1,7 +1,33 @@
-module.exports = {
-    testSocketConnection: (client) => {
-        client.on('test-conn', (cb) => {
-            cb('hello world');
-        });
+module.exports = (socket) => ({
+    handleNewRun: (data, callback) => {
```

Comment:
Why do we have `callback` here if we're emitting on socket? Aren't these even handlers "fire-and-forget"?

Reply:
These are just acknowledgments, clients can use to validate that their request and args have reached the broker's socket layer by examining this callback.

Reply:
Are we using these in any way? I don't think this is required because most of the events would cause another event to be fired.
E.g. when you pause a run, the `pause` callback is called.

Reply:
This is used for testing, if you look at the unit tests they are validating the callback args to ensure that the socket layer has accepted their emits.